src/demos/vdemo.rs

```diff
@@ -568,13 +571,23 @@ pub fn random_quorum_certificate<TYPES: NodeType, LEAF: LeafType<NodeType = TYPE
 }
 
 /// Provides a random [`ValidatingLeaf`]
-pub fn random_validating_leaf<TYPES: NodeType<ConsensusType = ValidatingConsensus>>(
+/// FIXME this is morally wrong Can't think of another way to do this...
```

Comment:
The trait bounds look reasonable to me. Is there anything that you think is wrong?

Reply:
I had made this function dependent on `NodeImplementation` as a generic. That didn't make sense to me since it's only about a leaf. Turns out we can drop that as well as the trait bound I had added!